flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/ContaineredTaskManagerParameters.java

```diff
@@ -48,13 +48,17 @@
 	/** Environment variables to add to the Java process. */
 	private final HashMap<String, String> taskManagerEnv;
 
+	private final TaskExecutorResourceSpec taskExecutorResourceSpec;
```

Comment:
I don't think the legacy fields can be mapped to the new ones.
`taskManagerHeapSizeMB` does not map to `taskHeapSize` (I know the names sounds similar and thus confusing), it's similar but also not exactly the same as `totalFlinkMemory`.

Mapping between the legacy and new fields could be complicated and against intuition. Therefore I'm in favor of keep the two whole sets of parameters for the legacy and new mode, so it would be easier for implementation and later removing legacy codes.

I'll add the `@nullable` annotations.